### DISCUSSION TITLE
how can I keep the & from changing?

### DISCUSSION THREAD

**Discussion by yolio2003** [remarkjs/remark#1074] [Q&A] [answered]
just want to do somthing in my custom Plugin,
but i found that i can't disable remark to modify my content  (i have read a lot of disscussion T_T):

from:

```markdown
[sdf](./architecture.md?a=b&c=d)
```

to: 
```markdown
[sdf](./architecture.md?a=b\&c=d)
```

this is the code i have try:
```typescript
const file = await unified()
    .use(remarkParse)
    // .use(customPlugin)
    .use(remarkStringify, {
      bullet: '-',
      fences: true,
      listItemIndent: 'one',
      incrementListMarker: undefined,
      // unsafe: [{ character: '&', after: '&', inConstruct: undefined }],
      // unsafe: undefined,
    })
    .process(fileContent);
```

I found that prettier can format `\&` back to `&`, which is so strange.

So in the end how can I keep the `&` from changing?
infact, i didn't want anything to be changed after remark.

thanks for any help!

**Comment by wooorm** [comment, marked as answer]
> infact, i didn't want anything to be changed after remark.

Impossible. remark will format your markdown.

> I found that prettier can format `\&` back to `&`, which is so strange.

It’s not. Prettier only looks at code you wrote. We allow code injected by plugins.

**Reply by yolio2003** [reply]
thank you very much, im confirmed now.
saldy i need to write a parser myself

**Reply by wooorm** [reply]
Why not a) spend 30 minutes looking through existing issues, as this question has been asked before, b) spend 30 minutes framing your question with more information, about what you exactly want, instead of c) spending 3+ months on building an alternative to unified?
